src/main/java/com/amihaiemil/docker/ListedContainers.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2018-2019, Mihai Emil Andronache
+ * All rights reserved.
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 1)Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2)Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * 3)Neither the name of docker-java-api nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.amihaiemil.docker;
+
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Listed containers, which may have filters applied.
+ * @author Michael Lux (michi.lux@gmail.com)
+ * @version $Id$
+ * @since 0.0.11
+ */
+final class ListedContainers extends RtContainers {
```

Comment:
@milux Take a look to the comments above; after they are implemented this class may be not necessary and those behaviors can come back to `RtContainers`

Reply:
Sure, we could move this functionality to `RtContainers`, but this is against the **good** pattern that I found with similar APIs in that library, where Sub-API-calls go to an abstract base class whereas concrete implementations for filter etc. go to a `ListedXYZ` class.
Do you really consider this a good idea?

Reply:
You're aware that interfaces do only define a contract/interface (thus their name...) and cannot contain any implementation?
Interface-like constructs that contain actual implementations, best known as traits, or other concepts for multiple inheritance are not available in Java.
We could move to project to Kotlin, though, then there are traits. :wink:

Reply:
And moving `iterator` and `all` to a base class that's not aware of filters and flags is not possible for the reasons explained above, unless you start to copy-paste the code to multiple levels of inheritance (or nesting in case of decorators) which is a terrible idea in terms of software design.